File: java/proxyPattern/RealLogo.java
package proxyPattern;

public class RealLogo implements Logo {
	private String logo;

	public RealLogo(String logo) {
		this.logo = logo;
		System.out.println("Logo " + this.logo + " is loading......");
	}

	@Override
	public void displayLogo() {
		System.out.println("Logo: " + this.logo);
	}

}

File: php/proxyPattern/RealLogo.php
<?php namespace php\proxyPattern;

class RealLogo implements Logo {
	/**
	 * Logo instance 
	 * 
	 * @var $logo
	 */
	private $logo;

	public function __construct($logo) {
		$this->logo = $logo;
		echo "Logo loading:............<br> ";
	}
	
	/**
	 * Logo display
	 * 
	 * @return void
	 */
	public function displayLogo() {
		printf("Logo: %s <br>",$this->logo);
	}
}
